tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -305,6 +361,143 @@ int main(int argc, char **argv)
         }
     }
 
+    /* s2n_tls13_server_nst_recv */
+    {
+        /**
+         *= https://tools.ietf.org/rfc/rfc8448#section-3
+         *# expanded (32 octets):  4e cd 0e b6 ec 3b 4d 87 f5 d6 02 8f 92 2c
+         *# a4 c5 85 1a 27 7f d4 13 11 c9 e6 2d 2c 94 92 e1 c4 f3
+         **/
+         S2N_BLOB_FROM_HEX(expected_session_secret,
+         "4e cd 0e b6 ec 3b 4d 87 f5 d6 02 8f 92 2c \
+            a4 c5 85 1a 27 7f d4 13 11 c9 e6 2d 2c 94 92 e1 c4 f3");
+
+        uint8_t test_ticket[] = { TEST_TICKET };
+        uint8_t test_ticket_age_add[] = { TEST_TICKET_AGE_ADD };
+        uint8_t nst_data[] = {
+            0x01, 0x01, 0x01, 0x01, /* ticket lifetime */
+            TEST_TICKET_AGE_ADD,    /* ticket age add */
+            0x02,                   /* nonce len */
+            0x00, 0x00,             /* nonce */
+            0x00, 0x03,             /* ticket len */
+            TEST_TICKET,            /* ticket */
+            0x00, 0x00,             /* extensions len */
+        };
+
+        /* Reads all written ticket data and tests session_ticket_cb serialized session data */
+        {
```

Comment:
This test is very long, and the small set of values we care about are getting lost in the noise. I'd clearly mark them, probably with little curl-braced "sub tests", even though they're coming from the same input.

The other issue is that the inputs to this test mix known values from the handshake trace RFC with arbitrary data. If you're going to do a known value test, do it with the exact NST message from the RFC.